auto_install/install.sh

```diff
@@ -137,350 +149,512 @@ main(){
 	"${opt1a}"  "${opt1b}" \
 	"${opt2a}"  "${opt2b}" \
-	"${opt3a}"  "${opt3b}" 3>&2 2>&1 1>&3) || \
-	{ echo "::: Cancel selected. Exiting"; exit 1; }
+	"${opt3a}"  "${opt3b}" 3> /dev/stderr 2> /dev/stdout >&3)
+	then
+		echo '::: Cancel selected. Exiting'
+		exit 1
+	fi
 
 	echo "::: ${UpdateCmd} option selected."
 }
 
-
-distroCheck(){
+distroCheck() {
 	# Check for supported distribution
 	# Compatibility, functions to check for supported OS
 	# distroCheck, maybeOSSupport, noOSSupport
 	# if lsb_release command is on their system
-	if command -v lsb_release > /dev/null; then
-
+	if command -v lsb_release > /dev/null
+	then
 		PLAT=$(lsb_release -si)
 		OSCN=$(lsb_release -sc)
-
 	else # else get info from os-release
+		declare -A VER_MAP
+
+		VER_MAP=(['9']='stretch')
+		VER_MAP+=(['10']='buster')
+		VER_MAP+=(['11']='bullseye')
+		VER_MAP+=(['16.04']='xenial')
+		VER_MAP+=(['18.04']='bionic')
+		VER_MAP+=(['20.04']='focal')
 
 		# shellcheck disable=SC1091
 		source /etc/os-release
+
 		PLAT=$(awk '{print $1}' <<< "$NAME")
-		VER="$VERSION_ID"
-		declare -A VER_MAP=(["9"]="stretch" ["10"]="buster" ["11"]="bullseye" ["16.04"]="xenial" ["18.04"]="bionic" ["20.04"]="focal")
+		VER="${VERSION_ID}"
+
 		OSCN=${VER_MAP["${VER}"]}
+
+		# Alpine support
+		[ -z "${OSCN}" ] && \
+			OSCN="${VER}"
 	fi
 
-	case ${PLAT} in
-		Debian|Raspbian|Ubuntu)
-			case ${OSCN} in
-				stretch|buster|bullseye|xenial|bionic|focal)
-				:
-				;;
+	case "${PLAT}" in
+		Debian | Raspbian | Ubuntu)
+			case "${OSCN}" in
+				stretch | buster | bullseye | xenial | bionic | focal)
+					:
+					;;
 				*)
-				maybeOSSupport
-				;;
+					maybeOSSupport
+					;;
 			esac
-		;;
+			;;
+		Alpine)
+			PKG_MANAGER='apk'
+			UPDATE_PKG_CACHE="${PKG_MANAGER} update; ${PKG_MANAGER} upgrade --prune"
+			PKG_INSTALL="${PKG_MANAGER} --no-cache add"
+			PKG_COUNT="${PKG_MANAGER} --no-cache upgrade -s | grep -sEe ' [0-9]+ packages' | sed -Ee 's/.* ([0-9]+) packages/\1/' || true"
+			CHECK_PKG_INSTALLED="${PKG_MANAGER} --no-cache info -e"
+
+			# substitute pakcges with its Alpine version
+			BASE_DEPS=($(echo "${BASE_DEPS[@]}" | \
+				sed -Ee 's/dnsutils/bind\-tools/'))
+			BASE_DEPS=($(echo "${BASE_DEPS[@]}" | \
+				sed -Ee 's/whiptail/newt/'))
+
+			# remove useless (for Alpine) packages
+			BASE_DEPS=($(echo "${BASE_DEPS[@]}" | \
+				sed -Ee 's/grepcidr //'))
+			BASE_DEPS=($(echo "${BASE_DEPS[@]}" | \
+				sed -Ee 's/bsdmainutils //'))
+
+			# add other dependencies
+			## install some utilities
+			BASE_DEPS+=(coreutils util-linux wget openssl)
+			## install docs and manuals
+			BASE_DEPS+=(docs man-pages mandoc mandoc-apropos)
+			## install a services manager
+			BASE_DEPS+=(openrc)
+
+			# install grepcidr manually
+			## install dependencies
+			"${PKG_INSTALL}" build-base make
+
+			## download binaeries
+			wget -o /dev/null -O grepcidr.tar.gz --no-cache http://www.pc-tools.net/files/unix/grepcidr-2.0.tar.gz
```

Comment:
Use `curl` here, so no additional dependency is required:
```suggestion
			curl -fLo grepcidr.tar.gz 'http://www.pc-tools.net/files/unix/grepcidr-2.0.tar.gz'
```

Reply:
It's the source code of grepcidr
I don't found other mirrors